src/util/duration.cpp

```diff
@@ -26,8 +27,8 @@ QChar DurationBase::kDecimalSeparator = QChar(0x002E);
 
 // static
 QString DurationBase::formatTime(double dSeconds, Precision precision) {
-    if (dSeconds < 0.0) {
-        // negative durations are not supported
+    if (dSeconds < 0.0 || !isfinite(dSeconds)) {
```

Comment:
I'm not sure if those functions in fpclassify.h actually work since we are compiling with `-ffast-math`. This was my experience when fixing the invalid bpm values. Forgot to note it down somewhere.

Reply:
It works. A unit test is already in place: 
https://github.com/mixxxdj/mixxx/blob/f4dc2318e8819bd3cef81caf7bf15403b6fcd2c7/src/test/durationutiltest.cpp#L117